### PR TITLE
Convert tests to assertj; Re-use DataTypeTestCase

### PR DESCRIPTION
**Pull request by mfussenegger**
See commits

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java [hunk 1]
```diff
@@ -126,12 +127,16 @@ public void createFields(Consumer<IndexableField> onField,
...
                                      boolean docValued,
                                      boolean stored) {
-                if (indexed) {
-                    onField.accept(new FloatPoint(name, value.floatValue()));
-                }
-                if (docValued) {
-                    onField.accept(new SortedNumericDocValuesField(name,
-                        NumericUtils.floatToSortableInt(value.floatValue())));
+                if (indexed && docValued) {
+                    onField.accept(new FloatField(name, value.floatValue(), Store.NO));
```
Comment (mfussenegger) [inline server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java:131]: Required to pass https://github.com/crate/crate/blob/b84fd0ffbbc3570b08b91e3adb07f6c4e0ccf9b3/server/src/testFixtures/java/io/crate/types/DataTypeTestCase.java#L133

`ValueIndexer` and DocMapper parser must result in the same amount of fields

File: server/src/test/java/io/crate/DataTypeTest.java [hunk 2]
```diff
@@ -117,63 +89,65 @@ public void testForValueWithObjectList() {
...
+        assertThat(dataType).isEqualTo(new ArrayType<>(DataTypes.FLOAT));
     }
 
     @Test
     public void testForValueWithEmptyList() {
         List<Object> objects = Arrays.<Object>asList();
-        DataType type = DataTypes.guessType(objects);
-        assertEquals(type, new ArrayType(DataTypes.UNDEFINED));
+        DataType<?> type = DataTypes.guessType(objects);
+        assertEquals(type, new ArrayType<>(DataTypes.UNDEFINED));
```
Comment (matriv) [inline server/src/test/java/io/crate/DataTypeTest.java:129]: missed assertj assertion?

File: server/src/test/java/io/crate/types/ArrayTypeTest.java [hunk 3]
```diff
@@ -21,71 +21,39 @@
...
-        BytesStreamOutput out = new BytesStreamOutput();
-        streamer.writeValueTo(out, serArray);
-
-        StreamInput in = out.bytes().streamInput();
-
-        assertThat(streamer.readValueFrom(in), is(serArray));
+    public void test_pg_string_array_literal_can_be_converted_to_values() {
+        ArrayType<String> strArray = new ArrayType<>(DataTypes.STRING);
+        List<String> values = strArray.implicitCast("{a,abc,A,ABC,null,\"null\",NULL,\"NULL\"}");
+        assertThat(values).contains("a", "abc", "A", "ABC", null, "null", null, "NULL");
```
Comment (matriv) [inline server/src/test/java/io/crate/types/ArrayTypeTest.java:56]:
```suggestion
        assertThat(values).containsExactly("a", "abc", "A", "ABC", null, "null", null, "NULL");
```

**Review by matriv — APPROVED** [review]
Thx, left 2 minor comments.

**Pull request merged**